### PR TITLE
add vista2d

### PR DESCRIPTION
**Pull request by yiheng-wang-nv**
Fixes # .

### Description
A few sentences describing the changes proposed in this pull request.

### Status
**Ready/Work in progress/Hold**

### Please ensure all the checkboxes:

- [x] Codeformat tests passed locally by running `./runtests.sh --codeformat`.
- [ ] In-line docstrings updated.
- [ ] Update `version` and `changelog` in `metadata.json` if changing an existing bundle.
- [ ] Please ensure the naming rules in config files meet our requirements (please refer to: `CONTRIBUTING.md`).
- [ ] Ensure versions of packages such as `monai`, `pytorch` and `numpy` are correct in `metadata.json`.
- [ ] Descriptions should be consistent with the content, such as `eval_metrics` of the provided weights and TorchScript modules.
- [ ] Files larger than 25MB are excluded and replaced by providing download links in `large_file.yml`.
- [ ] Avoid using path that contains personal information within config files (such as use `/home/your_name/` for `"bundle_root"`).

### CODE REVIEW

File: models/vista2d/scripts/cell_sam_wrapper.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+from segment_anything.build_sam import build_sam_vit_b
+from torch import nn
+from torch.nn import functional as F
+
+
+class CellSamWrapper(torch.nn.Module):
```
Comment (KumoLiu) [inline models/vista2d/scripts/cell_sam_wrapper.py:18]: I guess we can use the component in the core here? cc @myron

Comment (yiheng-wang-nv) [reply models/vista2d/scripts/cell_sam_wrapper.py:18]: Hi @KumoLiu , are all components migrated into MONAI now? If not, I would suggest to use the existing version (like vista3d and maisi)

Comment (KumoLiu) [reply models/vista2d/scripts/cell_sam_wrapper.py:18]: If this one is just a copy, we can merge this one first. Then refactor it using the oss components (the same to maisi and vista3d). The refactor should also be done in 1.4.

**Comment by Nic-Ma** [comment]
I think we have too many datalists in this PR, it's more than 180k lines..If we put the same things in a tutorial later, I feel it's not acceptable.
Can we refer to the MAISI bundle?
https://github.com/Project-MONAI/model-zoo/blob/dev/models/maisi_ct_generative/large_files.yml#L16

Thanks.

**Comment by mingxin-zheng** [comment]
> I think we have too many datalists in this PR, it's more than 180k lines..If we put the same things in a tutorial later, I feel it's not acceptable. Can we refer to the MAISI bundle? https://github.com/Project-MONAI/model-zoo/blob/dev/models/maisi_ct_generative/large_files.yml#L16
> 
> Thanks.

I agree with @Nic-Ma because
- Long data lists can be more prone to merge conflicts, especially if multiple contributors are updating the same data sets. 
- Redundancy Issues: if MONAI tutorials need to use these datalists, it has to either make a copy there, or depends on raw github links that point to the model zoo repo, which is not the preferred way of maintaining these repos.

File: models/vista2d/download_preprocessor/generate_json.py [hunk 2]
```diff
@@ -0,0 +1,982 @@
+import argparse
```
Comment (mingxin-zheng) [inline models/vista2d/download_preprocessor/generate_json.py:12]: License header is needed here

File: models/vista2d/configs/metadata.json [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        "0.0.8": "let pretrained_ckpt_name refer to ckpt_path",
+        "0.0.7": "update workflow input arg",
+        "0.0.6": "update the transforms and model, partly verified the reimplementation",
+        "0.0.5": "add inference logic and rename the entry point class from train.py to workflow.py",
+        "0.0.4": "update loadimage and weighted sampler",
+        "0.0.3": "fix final val error if override datalist",
+        "0.0.2": "fix attr error",
+        "0.0.1": "initialize the model package structure"
+    },
+    "monai_version": "1.3.1",
```
Comment (mingxin-zheng) [inline models/vista2d/configs/metadata.json]: I guess this needs to be "1.4"

Comment (yiheng-wang-nv) [reply models/vista2d/configs/metadata.json]: Hi @mingxin-zheng . 1.3.1 is fine here. It's the initial version with customized components

File: models/vista2d/download_preprocessor/all_file_downloader.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
+import argparse
```
Comment (mingxin-zheng) [inline models/vista2d/download_preprocessor/all_file_downloader.py:12]: license header needed here

File: models/vista2d/download_preprocessor/process_data.py [hunk 5]
```diff
@@ -0,0 +1,388 @@
+import argparse
```
Comment (mingxin-zheng) [inline models/vista2d/download_preprocessor/process_data.py:12]: license header needed here

**Comment by yiheng-wang-nv** [comment]
For the blossom CI issue, need to enable opencv lib in the dockerfile.

File: models/vista2d/docs/README.md [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+
+<div align="center"> <img src="https://developer-blogs.nvidia.com/wp-content/uploads/2024/04/vista-2d-model-precision-versus-specialist-model-baseline-performance.png" width="800"/> </div>
+
+
+
+### Install dependencies
+
+```
+pip install monai fire tifffile imagecodecs pillow fastremap
+pip install --no-deps cellpose natsort roifile
```
Comment (mingxin-zheng) [inline models/vista2d/docs/README.md]: Hi @KumoLiu , we have some new dependency here.

Comment (yiheng-wang-nv) [reply models/vista2d/docs/README.md]: Hi @mingxin-zheng , I remove this section and update metadata.json.
If specifying suitable version, all these dependencies can be installed with `pip install` (no need --no-deps) directly

File: models/vista2d/configs/metadata.json (hunk 3 above)
Comment (yiheng-wang-nv) [reply models/vista2d/configs/metadata.json]: Hi @mingxin-zheng , when testing with CI dockerfile, I used back to monai 1.4, and update all other dependencies. Thanks for the suggestion

File: models/vista2d/scripts/workflow.py [hunk 7]
```diff
@@ -0,0 +1,1205 @@
...
+    from cell_distributed_weighted_sampler import DistributedWeightedSampler
+    from components import LabelsToFlows, LoadTiffd, LogitsToLabels
+    from utils import LOGGING_CONFIG, parsing_bundle_config  # type: ignore
+else:
+    from .cell_distributed_weighted_sampler import DistributedWeightedSampler
+    from .components import LabelsToFlows, LoadTiffd, LogitsToLabels
+    from .utils import LOGGING_CONFIG, parsing_bundle_config
+
+
+logger = get_logger("VistaCell")
```
Comment (mingxin-zheng) [inline models/vista2d/scripts/workflow.py:60]: Hi @yiheng-wang-nv , can we change the `print` in this file to `logger.[info/error]`?

File: models/vista2d/scripts/workflow.py [hunk 8]
```diff
@@ -0,0 +1,1205 @@
...
+        data_list_files = dataset_data["data_list_files"]
+
+        if isinstance(data_list_files, str):
+            data_list_files = ConfigParser.load_config_file(
+                data_list_files
+            )  # if it's a path to a separate file with a list of datasets
+        else:
+            data_list_files = ensure_tuple(data_list_files)
+
+        if self.global_rank == 0:
```
Comment (mingxin-zheng) [inline models/vista2d/scripts/workflow.py:266]: We can apply RankFilter to the logger

File: models/vista2d/scripts/workflow.py [hunk 9]
```diff
@@ -0,0 +1,1205 @@
...
+
+if __name__ == "__main__":
+    # to be able to run directly as python scripts/workflow.py --config_file=...
+    # for debugging and development
+
+    from pathlib import Path
+
+    sys.path.append(str(Path(__file__).parent.parent))
+
+    # from scripts import *
```
Comment (mingxin-zheng) [inline models/vista2d/scripts/workflow.py:1199]: Can we remove this debugging line?

File: models/vista2d/scripts/workflow.py (hunk 7 above)
Comment (mingxin-zheng) [reply models/vista2d/scripts/workflow.py:60]: Closing comments as discussed offline.

**Comment by yiheng-wang-nv** [comment]
Discussed with @mingxin-zheng offline, for suggestions that may impact the bundle's internal logics, we will connect to the original author: @myron , and decide if we need to change in next version.
For this version, we will not modify any logics and only fix code format issues or other obvious bugs.

Although this PR is submitted by myself, the bundle content is totally copied from the original author side.

**Review by mingxin-zheng — APPROVED** [review]

**Pull request merged**